### ISSUE TITLE
3.5 issue in yaml.load_file

### ISSUE THREAD

**Issue by ldecicco-USGS** [viking/r-yaml#60]
When upgrading from R 3.4 to 3.5, we've seemed to lose our ability to use `yaml.load_file`:

R 3.5:
```
library(yaml)
x <- "https://raw.githubusercontent.com/USGS-VIZLAB/CIDA-Viz/master/viz.yaml"
yaml.load_file(x)
NULL
sessionInfo()
R version 3.5.0 Patched (2018-05-03 r74699)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods  
[7] base     

other attached packages:
[1] yaml_2.1.19

loaded via a namespace (and not attached):
[1] compiler_3.5.0 tools_3.5.0  
```
vs 3.4:
```
library(yaml)
x <- "https://raw.githubusercontent.com/USGS-VIZLAB/CIDA-Viz/master/viz.yaml"
yaml.load_file(x)
$vizlab
[1] "0.1.5"

$info
$info$id
[1] "ca-drought"

$info$name
[1] "California drought viz"

$info$date
[1] "2016-12-02"
...
sessionInfo()
R version 3.4.4 (2018-03-15)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods  
[7] base     

other attached packages:
[1] yaml_2.1.19

loaded via a namespace (and not attached):
[1] compiler_3.4.4 tools_3.4.4   
```

**Comment by viking** [comment]
Thanks for your report. I'll check it out.

**Comment by viking** [comment]
For the time being, do this instead:

```r
x <- url("https://raw.githubusercontent.com/USGS-VIZLAB/CIDA-Viz/master/viz.yaml")
yaml.load_file(x)
```

**Comment by viking** [comment]
This is happening because R's underlying connection behavior changed. The `yaml.load_file` function uses the `file()` function to try to open a filename. Apparently in 3.4, it accepted URLs, but in 3.5, it doesn't. I need to think about if I want to handle URLs specifically or require that people use the `url()` connection to pass into `yaml.load_file`.

**Comment by viking** [comment]
Since the `readLines()` function accepts either, I think it's reasonable that the `yaml.load_file` function does as well.

**Comment by ldecicco-USGS** [comment]
Good sleuthing! That might actually explain another issue I just discovered in a completely unrelated project, so double thanks.

I'm changing my code to:
```
yaml.load_file(url(x))
```
and that will be fine for me.

**Comment by viking** [comment]
It turns out that the problem isn't the `file()` function, it's specifying the encoding. This works:
```r
yaml.load_file(file(urlString))
```

This doesn't:
```r
yaml.load_file(file(urlString, encoding = "UTF-8"))
```

The latter is what `yaml.load_file` does when you pass in a string instead of a connection. I'm not sure if this is a problem with your specific YAML file or if this is a problem in general. I'll keep looking.

**Comment by viking** [comment]
Looks like this is a bug in R itself. I just checked the development version of R and it seems to have been fixed. I'll leave this issue open until the new R release and re-check.

**Comment by viking** [comment]
This seems to be fixed in 3.5.1. Let me know if you have any trouble with it.
